test/browser/test/snapshot.test.ts

```diff
@@ -4,6 +4,16 @@ test('inline snapshot', () => {
   expect(1).toMatchInlineSnapshot('1')
 })
 
-test('file snapshot', () => {
+test('snapshot', () => {
   expect(1).toMatchSnapshot()
 })
+
+test('file snapshot', async () => {
+  await expect('my snapshot content')
+    .toMatchFileSnapshot('./__snapshots__/custom/my_snapshot')
+})
+
+test('file snapshot absent', async () => {
+  await expect(null)
+    .toMatchFileSnapshot('./__snapshots__/custom/my_snapshot_absent')
```

Comment:
This looks weird. Why doesn't it generate a snapshot?

Reply:
This test is (maybe too indirectly) validate that `readSnapshotFile()` is returning `null` when the file doesn't exist. `./__snapshots__/custom/my_snapshot_absent` isn't an existing snapshot, so it matches against `null` (with the newly added assert logic).

Reply:
It should always generate a new snapshot

Reply:
Since I reverted the behaviour change in `packages/snapshot/src/port/state.ts` that facilitated the test, I'll remove this test. In the context of this PR, it was anyway testing beyond the change scope.